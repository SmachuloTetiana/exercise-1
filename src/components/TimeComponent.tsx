import React, { useEffect, useState } from "react";
import "./style.css";

const TimeComponent: React.FC = () => {
    const [time, setTime] = useState(new Date().toLocaleTimeString());
    let interval: ReturnType<typeof setInterval>;

    useEffect(() => {
        interval = setInterval(() => updateTime(), 1000);

        return () => clearInterval(interval);
    }, []);

    const updateTime = () => {
        setTime(new Date().toLocaleTimeString());
    };

    return (
        <main className="container">
            <h1 className="time">{time}</h1>
        </main>
    );
};

export default TimeComponent;
