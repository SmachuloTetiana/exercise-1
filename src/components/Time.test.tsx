import React from "react";
import {shallow} from "enzyme";
import TimeComponent from "./Time";

describe("Show time", () => {
    test("should render", () => {
        const component = shallow(<TimeComponent />);

        expect(component).toMatchSnapshot();
    });
});
