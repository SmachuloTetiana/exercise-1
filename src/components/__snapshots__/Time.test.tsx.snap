// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show time should render 1`] = `ShallowWrapper {}`;
